### PR TITLE
chore(codecov): Fix layout and required changes

### PR DESCRIPTION
**Pull request by armenzg**
Changes:

- Undo some of my changes in #75644
- Only show comment when coverage drops (not when it increases) and we have uncovered lines

### CODE REVIEW

File: codecov.yml [hunk 1]
```diff
@@ -75,21 +75,26 @@ flag_management:
 
 # Read more here: https://docs.codecov.com/docs/pull-request-comments
 comment:
+  #### FOR TESTING ####
+  # Uncomment this if you want to test commenting configurations changes
+  require_changes: false
+  require_bundle_changes: false
```
Comment (armenzg) [inline codecov.yml:81]: I will comment out these two lines once I'm done testing this PR.

File: codecov.yml [hunk 2]
```diff
@@ -75,21 +75,26 @@ flag_management:
...
+  # Show the git diff and condensed files
+  layout: 'diff, condensed_files'
+  # Do not show the project coverage in the comment
   hide_project_coverage: true
   # Update, if exists. Otherwise post new. Skip if deleted.
   behavior: once
-  # Comments will only post when coverage changes. Furthermore, if a comment
-  # already exists, and a newer commit results in no coverage change for the
-  # entire pull, the comment will be deleted.
-  require_changes: true
```
Comment (armenzg) [inline codecov.yml:89]: Moved to the top section.

File: codecov.yml [hunk 3]
```diff
@@ -75,21 +75,26 @@ flag_management:
...
   require_base: true # must have a base report to post
   require_head: true # must have a head report to post
-  require_bundle_changes: True # only post a comment when there are bundle size changes
```
Comment (armenzg) [inline codecov.yml:92]: Moved to the top section.

File: codecov.yml [hunk 4]
```diff
@@ -75,21 +75,28 @@ flag_management:
 
 # Read more here: https://docs.codecov.com/docs/pull-request-comments
 comment:
+  #### FOR TESTING ####
+  # If you want to test in your PR commenting configurations changes,
+  # comment out this line and uncomment the next line in the next section
+  # require_bundle_changes: false
+  ######################
+  # Post comment if there are changes in bundle size increases greater than value specified
+  require_bundle_changes: 'bundle_increase'
```
Comment (armenzg) [inline codecov.yml:84]: Just moving it up here from the bottom of this section.

File: codecov.yml [hunk 5]
```diff
@@ -75,21 +75,28 @@ flag_management:
...
   # This is the addition of carry forward builds and fresh builds, thus, it's the addition
   # of the FE and BE builds
   after_n_builds: 22
-  # Show only the git diff / patch coverage, and exclude any comparison between base and head:
-  layout: 'condensed_header, condensed_files, condensed_footer'
+  # What to render in the comment
+  layout: 'diff, condensed_files'
+  # Do not show the project coverage in the comment
   hide_project_coverage: true
+  show_carryforward_flags: true
```
Comment (armenzg) [inline codecov.yml:93]: This is new. It's explained [here](https://docs.codecov.com/docs/carryforward-flags#carryforward-flags-in-the-pull-request-comment).

It should show like this:
image

File: codecov.yml [hunk 6]
```diff
@@ -75,21 +75,28 @@ flag_management:
...
   # Update, if exists. Otherwise post new. Skip if deleted.
   behavior: once
-  # Comments will only post when coverage changes. Furthermore, if a comment
-  # already exists, and a newer commit results in no coverage change for the
-  # entire pull, the comment will be deleted.
-  require_changes: true
+  # Only post if coverage drops OR there are uncovered lines of code
+  require_changes: 'coverage_drop OR uncovered_patch'
```
Comment (armenzg) [inline codecov.yml:97]: Explained [here](https://docs.codecov.com/docs/pull-request-comments#requiring-changes).

> true - Only post comment if there are changes in coverage (positive or negative)

We don't care as much if the coverage increases.
> # Only post if coverage drops OR there are uncovered lines of code
> comment:
>    require_changes: "coverage_drop OR uncovered_patch"

File: codecov.yml (hunk 5 above)
Comment (armenzg) [inline codecov.yml:93]:
```suggestion
```

**Review by asottile-sentry — APPROVED** [review]

**Pull request merged**